File: tests/integration/aliases/python/step1/scenario_retype_component_child.py
# Copyright 2026, Pulumi Corporation.  All rights reserved.

import pulumi


class FooResource(pulumi.ComponentResource):
    def __init__(self, name, opts=None):
        super().__init__("my:module:FooResource", name, None, opts)


class ComponentResource(pulumi.ComponentResource):
    def __init__(self, name, opts=None):
        super().__init__("my:module:ComponentResource", name, None, opts)
        FooResource("childRetypeChild", pulumi.ResourceOptions(parent=self))


ComponentResource("compRetypeChild")

File: tests/integration/aliases/nodejs/step1/scenario_retype_component_child.ts
// Copyright 2026, Pulumi Corporation.  All rights reserved.

import * as pulumi from "@pulumi/pulumi";

class FooResource extends pulumi.ComponentResource {
    constructor(name: string, opts?: pulumi.ComponentResourceOptions) {
        super("my:module:FooResource", name, {}, opts);
    }
}

class ComponentResource extends pulumi.ComponentResource {
    constructor(name: string, opts?: pulumi.ComponentResourceOptions) {
        super("my:module:ComponentResource", name, {}, opts);
        new FooResource("childRetypeChild", { parent: this });
    }
}

new ComponentResource("compRetypeChild");
